pyomo/contrib/doe/fim_doe.py

```diff
@@ -122,8 +176,9 @@ def __check_inputs(self, check_mode=False):
 
 
     def optimize_doe(self,  design_values, if_optimize=True, objective_option='det',
```

Comment:
Alternate design: this optionally takes a measurement object instead of `jac_involvement_measurement`. Or this functionality moves to the measurement class. I think we should isolate all of the logic associated with flattening inside the measurement class.

Reply:
Still need to update `optimize_doe`.